driver/driver.py

```diff
@@ -20,81 +9,266 @@
 
 from glob import glob
 from smartsim import Experiment
-from smartsim.settings import SbatchSettings, SrunSettings
-from smartsim.database import SlurmOrchestrator
-from smartsim.utils.log import log_to_file
-
-# logging output saved to file
-log_to_file("./driver.log")
-
-# experiment parameters
-DB_NODES = 16
-ENSEMBLE_NODES = 228
-ENSEMBLE_MEMBERS = 12
-MOM6_EXE = "/lus/cls01029/spartee/poseidon/NCAR_ML_EKE/MOM6/build/gnu/ice_ocean_SIS2/repro/MOM6"
-
-# name of experiment where output will be placed
-experiment = Experiment("AI-EKE-MOM6", launcher="slurm")
-
-# define slurm execution settings for a single member
-# of the ensemble
-mom_opts= {
-    "nodes": 19,
-    "ntasks": 910,
-    "exclusive": None,
-}
-mom_settings = SrunSettings(MOM6_EXE, run_args=mom_opts)
-
-# define batch parameters for entire ensemble
-batch_opts = {
-    "mincpus": 96,
-    "ntasks-per-node": 48,
-    "exclusive": None
-}
-ensemble_batch = SbatchSettings(nodes=ENSEMBLE_NODES, time="10:00:00", batch_args=batch_opts)
-
-# create reference to MOM6 ensemble
-mom_ensemble = experiment.create_ensemble("MOM",
-                                          batch_settings=ensemble_batch,
-                                          run_settings=mom_settings,
-                                          replicas=ENSEMBLE_MEMBERS)
-
-# Attach input files and configuration files to each
-# MOM6 simulation
-mom_ensemble.attach_generator_files(
-    to_configure=glob("../MOM6/MOM6_config/*"),
-    to_copy="../MOM6/OM4_025",
-    to_symlink="../MOM6/INPUT",
-)
-
-# configs to write into 'to_configure' files listed
-# above. If you change the number of processors for
-# each MOM6 simulation, you will need to change this.
-MOM6_config = {
-    "SIM_DAYS": 1, # length of simlations 
-    "DOMAIN_LAYOUT": "32,36",
-    "MASKTABLE": "mask_table.242.32x36"
+from smartsim.database import Orchestrator
+from smartsim.log import log_to_file
+
+
+def create_mom_ensemble( experiment, args ):
+    """Creates the ensemble object that stores all configuration options
+    """
+
+    mom6_batch_args = {
+        "C":args.ensemble_node_features,
```

Comment:
this is just about the only part of the script that will not work across all launchers. I would suggest changing the CLI to allow passing in ``--batch_arg=`arg=value``` or just define a structure at the top that contains these. that way it can at least be added to without removing this code.